server/src/main/java/org/elasticsearch/node/NodeConstruction.java

```diff
@@ -1038,6 +1040,7 @@ record PluginServiceInstances(
             b.bind(MetadataCreateIndexService.class).toInstance(metadataCreateIndexService);
             b.bind(MetadataUpdateSettingsService.class).toInstance(metadataUpdateSettingsService);
             b.bind(SearchService.class).toInstance(searchService);
+            b.bind(SearchTransportAPMMetrics.class).toInstance(searchTransportAPMMetrics);
```

Comment:
PR looks really good! I have only one question and it's about this line. I don't quite understand why are we add this here, is because we want the instance injected in the `TransportSeachAction` or am I missing something?

Reply:
Yep that is what binds the component and makes it available to those classes that use dependency injection (like transport actions)

Reply:
If we (re)make the registration static, as requested by @javanna [in this comment](https://github.com/elastic/elasticsearch/pull/102557#discussion_r1415376861), then we can no longer access `searchTransportAPMMetrics` as an instance member  of `TransportSeachAction` and instead need to provide it through dependency injection.